pkg/aggregator/sender.go

```diff
@@ -180,19 +180,25 @@ func (s *checkSender) sendMetricSample(
 	tags []string,
 	mType metrics.MetricType,
 	flushFirstValue bool,
-	noIndex bool) {
+	noIndex bool,
+	timestamp float64,
+) {
 	tags = append(tags, s.checkTags...)
 
 	log.Trace(mType.String(), " sample: ", metric, ": ", value, " for hostname: ", hostname, " tags: ", tags)
 
+	if timestamp == 0 {
```

Comment:
Open question: Should we have additional checks to prevent metrics to be submitted in the future/or too early in the past? How does dogstatsd handle that today? 
Because the intake will only accept metrics in a given timerange

Reply:
I think dogstatsd only checks that a timestamp is strictly above 0 ([see code](https://github.com/DataDog/datadog-go/blob/master/statsd/statsd.go#L711-L713))

Kept the same logic in this PR
https://github.com/DataDog/datadog-agent/blob/1960b04d9dbce5a5f37a60027a96c507be0ad4c3/pkg/aggregator/sender.go#L312-L314